tests/common/quantization/test_ignore_post_processing.py

```diff
@@ -0,0 +1,288 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import List
+
+import pytest
+
+from nncf.common.graph.layer_attributes import Dtype
+from nncf.common.quantization.quantizer_propagation.solver import PostprocessingNodeLocator
+from nncf.common.quantization.structs import QuantizableWeightedLayerNode
+from nncf.common.quantization.quantizer_propagation.graph import QuantizerPropagationStateGraph
+from nncf.common.quantization.structs import QuantizerConfig
+from nncf.common.graph.graph import NNCFGraph
+from nncf.common.graph.operator_metatypes import OutputNoopMetatype
+from nncf.common.graph.operator_metatypes import InputNoopMetatype
+from nncf.common.graph.definitions import NNCFGraphNodeType
+from nncf.common.insertion_point_graph import InsertionPointGraph
+from nncf.common.utils.registry import Registry
+
+from tests.common.quantization.metatypes import METATYPES_FOR_TEST
+from tests.common.quantization.metatypes import TopKTestMetatype
+from tests.common.quantization.metatypes import NMSTestMetatype
+from tests.common.quantization.metatypes import WEIGHT_LAYER_METATYPES
+
+from collections import Counter
+
+ALL_SYNTHETIC_NNCF_GRAPH = Registry('ONNX_SYNTHETIC_MODELS')
+
+
+class NodeWithType:
+    def __init__(self, name, op_type):
+        self.node_name = name
+        self.node_op_type = op_type
+
+
+class NNCFGraphToTest:
+    def __init__(self, nodes: List[NodeWithType], node_edges):
+        self.nncf_graph = NNCFGraph()
+        for node in nodes:
+            node_name, node_op_type = node.node_name, node.node_op_type
+            if 'Output' in node_name:
+                metatype = OutputNoopMetatype
+                node_op_type = NNCFGraphNodeType.OUTPUT_NODE
+            elif 'Input' in node_name:
+                metatype = InputNoopMetatype
+                node_op_type = NNCFGraphNodeType.INPUT_NODE
+            else:
+                metatype = METATYPES_FOR_TEST.get_operator_metatype_by_op_name(node_op_type)
```

Comment:
Instead of matching a string to a metatype here, why not just assign the metatype directly as a parameter to the `NodeWithType` constructor?

Reply:
Done